rq/job.py

```diff
@@ -1248,6 +1253,12 @@ def delete(
         if delete_dependents:
             self.delete_dependents(pipeline=pipeline)
 
+        if self.batch_id:
+            from .batch import Batch
+
+            batch = Batch.fetch(id=self.batch_id, connection=self.connection)
```

Comment:
This incurs an extra Redis call, let's make a `Batch.get_key(batch_id)` method so we can get the batch's Redis key without using `Batch.fetch()`.

Reply:
Added get_key method.